### PR TITLE
Refactor states

### PR DESCRIPTION
**Pull request by james-ward**
Make the state machine states self contained.

### CODE REVIEW

File: automations/winchautomation.py [hunk 1]
```diff
@@ -8,29 +8,29 @@ class WinchAutomation(StateMachine):
...
     @state(first=True)
-    def initialise_winch(self):
+    def catch_rope(self):
         self.put_dashboard()
-
         self.winch.piston_open()
         self.winch.rotate_winch(0.3)
-        self.next_state("climb")
+
+        if self.winch.on_rope_engaged() and state_tm > 2:
```
Comment (auscompgeek) [inline automations/winchautomation.py]: Shouldn't `state_tm` be listed as an argument for this state?

As an aside, we should probably have tests for the automations to catch these bugs…

File: autonomous/autonomous.py [hunk 2]
```diff
@@ -196,21 +185,18 @@ def rotate_towards_peg(self, initial_call):
...
-            self.next_state("drive_to_wall")
             self.manipulategear.engage()
-
-    @state
-    def drive_to_wall(self, initial_call):
         if not self.manipulategear.is_executing:
+            self.next_state("roll_back")
+    @state
+    def roll_back(self, initial_call):
+        if intial_call:
```
Comment (auscompgeek) [inline autonomous/autonomous.py]: I'm guessing this was a typo. 😜

File: autonomous/autonomous.py [hunk 3]
```diff
@@ -133,61 +111,72 @@ def measure_position(self, initial_call):
...
 
     @state
     def drive_align_segment(self, initial_call):
-        if not self.profilefollower.queue[0]:
-            rotate_towards_wall = generate_trapezoidal_trajectory(
-                    self.bno055.getHeading(), 0,
-                    self.perpendicular_heading, 0, self.rotate_velocity,
-                    self.rotate_accel_speed, -self.rotate_accel_speed/2)
-            self.profilefollower.modify_queue(heading=rotate_towards_wall, overwrite=True)
+        if intial_call:
```
Comment (auscompgeek) [inline autonomous/autonomous.py]: Meanwhile in the world of people who use completion… :wink:

File: automations/winchautomation.py (hunk 1 above)
Comment (james-ward) [inline automations/winchautomation.py]: Feel free to set that up.

**Pull request merged**